src/IO/IRemoteFileMetadata.cpp

```diff
@@ -0,0 +1,35 @@
+#include <IO/IRemoteFileMetadata.h>
+#include <Common/Exception.h>
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
+IRemoteFileMetadata::~IRemoteFileMetadata() {}
+
+RemoteFileMetadataFactory & RemoteFileMetadataFactory::instance()
+{
+    static RemoteFileMetadataFactory g_factory;
+    return g_factory;
+}
+
+IRemoteFileMetadataPtr RemoteFileMetadataFactory::createClass(const String & class_name)
```

Comment:
may be it is more like `get` instead of `createClass`? Also I think it should throw if not found.
```suggestion
IRemoteFileMetadataPtr RemoteFileMetadataFactory::get(const String & name)
```

Reply:
done